### ISSUE TITLE
production ready?

### ISSUE THREAD

**Issue by hiqbn** [dgryski/go-xxh3#1]
production ready?

**Comment by dgryski** [comment]
Yes and no.  While the base implementation matches the xxh3 results when it was first released, the hash algorithm itself is not finalized and further changes have been made upstream which are not reflected here.  Second, my implementation is only "lightly" optimized and is not taking advantage of the SIMD instructions that xxh3 function was designed to exploit. This would require an assembly implementation as the Go compiler does not generate these instructions. 

There is one at https://github.com/zeebo/xxh3 but I don't know if it has been updated to match the upstream version, which as I mentioned it still in flux.

**Comment by gitmko0** [comment]
xxh3 0.8 is out! quick implement it!

**Comment by seebs** [comment]
```
// may not be accurate for all inputs
func xxh3_0_8(in []byte) (out uint64) {
    return 0
}
```

hope this helps. if you find bugs feel free to suggest a patch that corrects them.
